python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -822,6 +822,49 @@ def test_write_batch_dedup(library_factory):
             assert data_key_version[s] == 0
 
 
+def test_read_batch_time_stamp(arctic_library):
+    """Should be able to write different size of batch of data."""
+    lib = arctic_library
+    sym = "sym_"
+    num_versions = 3
+    num_symbols = 3
+    for v_num in range(num_versions):
+        write_requests = []
+        for sym_num in range(num_symbols):
+            write_requests.append(
+                WritePayload(
+                    sym + str(sym_num), pd.DataFrame({"col": [sym_num + v_num, sym_num * v_num, sym_num - v_num]})
+                )
+            )
+        lib.write_batch(write_requests)
+        time.sleep(1)
+
+    data_non_batch = []
+    requests_batch = []
+    for sym_num in range(num_symbols):
+        versions = lib.list_versions(sym + str(sym_num))
+        for version_info in versions:
+            date_obj = versions[version_info].date
+            if date_obj.second == 59:
+                continue
+            as_of = datetime(
+                date_obj.year,
+                date_obj.month,
+                date_obj.day,
+                date_obj.hour,
+                date_obj.minute,
+                date_obj.second + 1,
+                tzinfo=timezone.utc,
+            )
+            data_non_batch.append(lib.read(sym + str(sym_num), as_of=as_of))
+            requests_batch.append(ReadRequest(sym + str(sym_num), as_of=as_of))
+
+    data_batch = lib.read_batch(requests_batch)
+    list_data_to_compate = list(zip(data_non_batch, data_batch))
+    for d1, d2 in list_data_to_compate:
+        assert_frame_equal(d1.data, d2.data)
```

Comment:
Can we also check that this is correct, not that it's the same behaviour as `read`.

Reply:
done the extra check

Reply:
Then what is the point of leaving the "is the same as `read`" check?

Reply:
yeh that's fair